internal/latestrelease/store.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2022 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package latestrelease
+
+import (
+	"bytes"
+	"time"
+
+	"github.com/mongodb/mongocli/internal/config"
+	"github.com/mongodb/mongocli/internal/file"
+	"github.com/mongodb/mongocli/internal/version"
+	"github.com/spf13/afero"
+)
+
+//go:generate mockgen -destination=../mocks/mock_version_store.go -package=mocks github.com/mongodb/mongocli/internal/latestrelease Store
+
+type State struct {
+	CheckedForUpdateAt   time.Time `yaml:"checked_for_update_at"`
+	LatestReleaseVersion string    `yaml:"latest_release"`
+}
+
+type BrewPath struct {
+	CheckedPathAt  time.Time `yaml:"checked_path_at"`
+	ExecutablePath string    `yaml:"path"`
+	HomePath       string    `yaml:"home_path"`
+}
+
+type Store interface {
```

Comment:
also does brew and latest version need to be in the same interface? they seem unrelated

Reply:
`Store` doesn't say much, look that we have a `Store` struct but never a struct interface, please follow best practices by naming your interface by what they do no what they are

Reply:
All right, just in case :) Updated both brew/release version stores to loadersavers. A bit new to me since I saw that we do that for 1-method interfaces then I thought we ended up in an "our call" situation, but I agree store does not say much.